shell/gpu/gpu_surface_metal.mm

```diff
@@ -85,23 +90,52 @@
 
     canvas->flush();
 
-    if (next_drawable_ == nullptr) {
-      FML_DLOG(ERROR) << "Could not acquire next Metal drawable from the SkSurface.";
+    GrMTLHandle drawable = next_drawable_;
```

Comment:
@chinmaygarde I tried to be clever by capturing the next_drawable_ in lambda, skia uses lazy proxies when we create the SKSurface from metal layer, which means that this variable can be null until canvas.flush() gets called. It only happens on real devices and not simulators for some reason hence this failed an actual device lab test during flutter roll. I've reverted to the earlier behavior.